util/encoding/float_test.go

```diff
@@ -172,8 +172,10 @@ func BenchmarkDecodeFloat(b *testing.B) {
 		vals[i] = EncodeFloat(nil, rng.Float64())
 	}
 
+	buf := make([]byte, 0, 100)
+
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		_, _ = DecodeFloat(vals[i%len(vals)])
+		_, _ = DecodeFloat(vals[i%len(vals)], buf)
```

Comment:
a more accurate benchmark would be a temporary buffer between DecodeFloat invocations. What does it look like then?